azure-proto-core/Resources/IdentityKind.cs

```diff
@@ -20,22 +18,52 @@ public IdentityKind(string kind)
         public string Value { get; private set; }
         public int CompareTo(IdentityKind other)
         {
-            throw new NotImplementedException();
+            bool? checkNull = checkBothNull(other.Value);
```

Comment:
I think this makes it more confusing than just having the logic inline here given the variable name "checknull" doesn't quite capture the full meaning.